apps/backend/src/datasources/QuestionaryDataSource.ts

```diff
@@ -14,9 +14,18 @@ export interface QuestionaryDataSource {
   ): Promise<AnswerBasic | null>;
   delete(questionary_id: number): Promise<Questionary>;
   getQuestionary(questionary_id: number): Promise<Questionary | null>;
-  getQuestionarySteps(questionaryId: number): Promise<QuestionaryStep[]>;
-  getBlankQuestionarySteps(templateId: number): Promise<QuestionaryStep[]>;
-  getBlankQuestionaryStepsByCallId(callId: number): Promise<QuestionaryStep[]>;
+  getQuestionarySteps(
+    questionaryId: number,
+    currentUserRole: string | null
```

Comment:
question: this is more a question if maybe a better approach is do the logic of filtering in the authorization layer, instead of sending it down to the database layer.

pros: 
- keeping db function signature simple and function body DRY 
- keeping authorization related code in the authorization layer
cons:
- slower to execute